runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -348,19 +385,61 @@ function LanguageTree:parse()
     end
   end
 
+  return query_time
+end
+
+--- Run a parse on the language tree.
+---
+--- If {range} isn't `false|nil`, then injections will be
+--- queried and parsed recursively.
+---
+---@param range boolean|Range|nil: Parse this range in the parsers source.
+---    This only affects regions in the source which have an explicit
+---    range defined which includes: regions set with `set_included_regions()`,
+---    and, injections.
```

Comment:
```suggestion
---@param range boolean|Range|nil: Parse this range in the parser's source.
---    This only affects regions in the source which have an explicit
---    range defined which includes: regions set with `set_included_regions()`
---    and injections.
```

This is a bit hard to parse for me (pun intended). Can you explain what exactly this `{range}` parameter controls, from a practical point of view?